Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1

```diff
@@ -171,13 +171,62 @@ param (
         Write-Verbose "Unable to get the path for net.exe. Net command will be 'net'"
         return 'net'
     }
+
+    function Create-DestinationDirectory(
+        [string]$path
+    )
+    {
+        $psCredentialObject = New-Object PSCredential($credential.UserName, (ConvertTo-SecureString $credential.Password -AsPlainText -Force))
+        $destPath = $path
+        $foundParentPath = $false
+        $isRoot = $false
+
+        Write-Verbose "Creating path to directory: $path"
+        while($destPath -and (-not $foundParentPath))
+        {
+            try
+            {
+                New-PSDrive -Name WFCPSDrive -PSProvider FileSystem -Root $destPath -Credential $psCredentialObject
+                $foundParentPath = $true
+                Write-Verbose "Found parent path"
+                $relativePath = $path.Substring($destPath.Length)
+                New-Item -ItemType Directory WFCPSDrive:$relativePath -Force
+                Write-Verbose "Created directory"
+                Remove-PSDrive -Name WFCPSDrive
+            }
+            catch 
+            {
+                Write-Verbose "Caught exception: $_.Exception.Message"
+                $parentPath = Split-Path -Path $destPath -Parent
+                if(($parentPath.Length -eq 0) -and ($isRoot -eq $false))
+                {
+                    $destPath = [IO.Path]::DirectorySeparatorChar + [IO.Path]::DirectorySeparatorChar + ([System.Uri]($destPath)).Host
+                    $isRoot = $true
+                    Write-Verbose "Check if root path exists: $destPath"
+                }
+                else
+                {
+                    $destPath = $parentPath
+                    Write-Verbose "Check if parent path exists: $destPath"
+                }
+            }
+        }
+    }
     
     $machineShare = Get-MachineShare -fqdn $fqdn -targetPath $targetPath    
     $destinationNetworkPath = Get-DestinationNetworkPath -targetPath $targetPath -machineShare $machineShare
+
+    Write-Verbose "machine share= $machineShare"
+    Write-Verbose "destination network path= $destinationNetworkPath"
     
     Validate-Credential $credential
     $userName = Get-DownLevelLogonName -fqdn $fqdn -userName $($credential.UserName)
     $password = $($credential.Password) 
+
+    if([bool]([uri]$targetPath).IsUnc)
```

Comment:
Why this check?

Reply:
I did not want to disturb the original working flow for non-UNC paths. In non-UNC scenarios the folder is anyways created by robocopy. So this fix is not required for that.

Reply:
As discussed, I believe we always deal with UNC after mapping the path.

Reply:
I'll check that. But even then, the current $targetPath is before mapping the path. This PR specifically aims to fix the case when the target path (before mapping) is a UNC path. That is what I am checking before calling Create-DestinationDirectory.